### PR TITLE
24953 - Update NoW Filing Page

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* /bcgov/entity#24953

*Description of changes:*
- Move the 2 POA check boxes to the first section + add the 2 properties in filing json
- Show the warning message and disable "Submit" when "POA has taken effect" is selected
- Change Court Order section to a common one
- Change the withdrawn filingId to a number (previously string)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by ArwenQin** [comment]
Updated first section, with 2 check boxes and the warning message:
image


Disable the "Submit" button when "POA has taken effect" is selected
image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-708-ofasgl8t.web.app

**Comment by ArwenQin** [comment]
Sample filing json

{
    "header": {
        "name": "noticeOfWithdrawal",
        "certifiedBy": "AQ",
        "email": "eason.pan@gov.bc.ca",
        "date": "2024-12-18",
        "documentOptionalEmail": "test@test.ca",
        "routingSlipNumber": "123456798",
        "priority": false
    },
    "business": {
        "foundingDate": "2024-12-10T23:37:44.000+00:00",
        "identifier": "BC0883809",
        "legalName": "0883809 B.C. LTD.",
        "legalType": "BC"
    },
    "noticeOfWithdrawal": {
        "filingId": 155118,
        "partOfPoa": false,
        "hasTakenEffect": false,
        "courtOrder": {
            "fileNumber": "TT0010045",
            "effectOfOrder": "planOfArrangement"
        }
    }
}

**Comment by ArwenQin** [comment]
Test: https://business-filings-dev--pr-708-ofasgl8t.web.app/BC0883809/notice-of-withdrawal?filingToBeWithdrawn=155118&filingId=0

The Save and Resume issue is not fixed yet

**Comment by severinbeauvais** [comment]
> Updated first section, with 2 check boxes and the warning message:

Looks great! Please increase the whitespace between the bottom of the warning box and the section frame. (Same whitespace all around.)

**Comment by severinbeauvais** [comment]
> Please increase the whitespace between the bottom of the warning box and the section frame. (Same whitespace all around.)

Please check the whitespace in the other sections, too. I see some inconsistencies.

Eg,
image

File: src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue [hunk 1]
```diff
@@ -62,32 +64,132 @@
           id="withdrawal-filing-date"
           cols="12"
           sm="9"
+          class="grey-text"
         >
           {{ getFormattedFilingDate() }}
         </v-col>
       </v-row>
+      <v-divider class="mt-2 mx-4" />
```
Comment (severinbeauvais) [inline src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue:72]: Don't be afraid to use whitespace between logical blocks (or around a divider) 😃

File: src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue [hunk 2]
```diff
@@ -102,5 +204,31 @@ export default class RecordToBeWithdrawn extends Mixins(DateMixin) {
...
+  @Watch('hasTakenEffect')
+  @Emit('hasTakenEffect')
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  emitEffect (hasTakenEffect: boolean): void {}
 }
 </script>
+
+<style scoped>
+.checkbox-label {
+  font-size: 14px;
```
Comment (severinbeauvais) [inline src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue:232]: We have a base class, `font-14`, to simplify cases like this :)

File: src/enums/effectOfOrderTypes.ts [hunk 3]
```diff
@@ -1,4 +1,3 @@
 export enum EffectOfOrderTypes {
-  PLAN_OF_ARRANGEMENT = 'planOfArrangement',
-  ARRANGEMENT_EFFECT = 'arrangementEffect'
+  PLAN_OF_ARRANGEMENT = 'planOfArrangement'
```
Comment (severinbeauvais) [inline src/enums/effectOfOrderTypes.ts:2]: Does this enum still make sense, since it has only 1 value?

Comment (ArwenQin) [reply src/enums/effectOfOrderTypes.ts:2]: This enum was there, before my previous changes. Other filings are using this too

File: src/views/NoticeOfWithdrawal.vue [hunk 4]
```diff
@@ -423,7 +432,7 @@ export default class NoticeOfWithdrawal extends Mixins(CommonMixin, DateMixin, F
       // this is the id of filing being withdrawn, and of THIS filing
       // if filingID is 0, this is a new filing
       // otherwise it's a draft filing
-      this.filingToBeWithdrawn = this.$route.query.filingToBeWithdrawn as string || ''
+      this.filingToBeWithdrawn = Number(this.$route.query.filingToBeWithdrawn) || null
```
Comment (severinbeauvais) [inline src/views/NoticeOfWithdrawal.vue:435]: Try this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus

Also, `NaN` is a special value to denote an invalid number, that you could also use if filingToBeWithdrawn isn't a route query parameter. (But no need to change here.)

File: tests/unit/NoticeOfWithdrawal.spec.ts [hunk 5]
```diff
@@ -77,7 +77,7 @@ describe('Notice of Withdrawal view', () => {
     expect(vm.filingData).not.toBeUndefined()
     expect(vm.filingData).not.toBeNull()
     expect(vm.filingData.length).toBe(1)
-    expect(vm.filingData[0].filingTypeCode).toBe('NOIWD')
+    expect(vm.filingData[0].filingTypeCode).toBe('NWITH')
```
Comment (severinbeauvais) [inline tests/unit/NoticeOfWithdrawal.spec.ts:80]: You can also an enum here if you want.

**Review by severinbeauvais — COMMENTED** [review]
Looks good so far.

Ask Jacqueline to have a look at this, in case she wants to tweak anything.

File: src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue (hunk 1 above)
Comment (ArwenQin) [reply src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue:73]: Updated

File: src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue (hunk 2 above)
Comment (ArwenQin) [reply src/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue]: Updated, thanks!

File: tests/unit/NoticeOfWithdrawal.spec.ts (hunk 5 above)
Comment (ArwenQin) [reply tests/unit/NoticeOfWithdrawal.spec.ts]: Yes, updated

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-708-ofasgl8t.web.app

**Comment by ArwenQin** [comment]
Updated the spaces around first section:
image

and Staff Payment
image


All sections are consistent now: py-8 px-5

File: src/views/NoticeOfWithdrawal.vue [hunk 6]
```diff
@@ -432,7 +431,7 @@ export default class NoticeOfWithdrawal extends Mixins(CommonMixin, DateMixin, F
       // this is the id of filing being withdrawn, and of THIS filing
       // if filingID is 0, this is a new filing
       // otherwise it's a draft filing
-      this.filingToBeWithdrawn = Number(this.$route.query.filingToBeWithdrawn) || null
+      this.filingToBeWithdrawn = +this.$route.query.filingToBeWithdrawn
```
Comment (severinbeauvais) [inline src/views/NoticeOfWithdrawal.vue:434]: Should you add a check that this is provided (same as filingId)? See lines 437-440.

File: src/views/NoticeOfWithdrawal.vue (hunk 4 above)
Comment (severinbeauvais) [reply src/views/NoticeOfWithdrawal.vue]: Looks OK!

**Review by severinbeauvais — COMMENTED** [review]
~This looks good to me, but I'm hoping you can fix the resume issue in this PR also.~ _fixed now, thanks!_

File: src/views/NoticeOfWithdrawal.vue [hunk 7]
```diff
@@ -423,11 +431,11 @@ export default class NoticeOfWithdrawal extends Mixins(CommonMixin, DateMixin, F
...
       // if filingID is 0, this is a new filing
       // otherwise it's a draft filing
-      this.filingToBeWithdrawn = this.$route.query.filingToBeWithdrawn as string || ''
+      this.filingToBeWithdrawn = +this.$route.query.filingToBeWithdrawn
       this.filingId = +this.$route.query.filingId // number or NaN
 
       // if required data isn't set, go back to dashboard
-      if (isNaN(this.filingId)) {
+      if (isNaN(this.filingId) || isNaN(this.filingToBeWithdrawn)) {
         this.navigateToBusinessDashboard(this.getIdentifier)
```
Comment (ArwenQin) [inline src/views/NoticeOfWithdrawal.vue:439]: add a check here

**Review by severinbeauvais — APPROVED** [review]
Pre-approved (assuming resume now works correctly). Good work! :)

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-708-ofasgl8t.web.app

**Review by eason-pan-bc — APPROVED** [review]
LGTM, I tried the link to resume the draft and it worked. (draft filing_id 155239) https://business-filings-dev--pr-708-ofasgl8t.web.app/BC0883809/notice-of-withdrawal?filingToBeWithdrawn=155118&filingId=155239

Update: the 'Resume' button on the dashboard still not working.

**Comment by ArwenQin** [comment]
Jacqueline has an update for the certify section:
image

I will update it now, and then we can merge it

**Comment by ArwenQin** [comment]
Jacqueline said it's not a requirement now, changed the design back. No updates needed. This PR is ready to merge.

**Review by meawong — APPROVED** [review]
Looks good to go!

**Pull request merged**